lib/ansible/modules/cloud/amazon/cloudformation.py

```diff
@@ -276,15 +280,16 @@
 from ansible.module_utils._text import to_bytes, to_native
 
 
-def get_stack_events(cfn, stack_name, token_filter=None):
+def get_stack_events(cfn, stack_name, events_limit, token_filter=None):
     '''This event data was never correct, it worked as a side effect. So the v2.3 format is different.'''
     ret = {'events': [], 'log': []}
 
     try:
         pg = cfn.get_paginator(
             'describe_stack_events'
         ).paginate(
-            StackName=stack_name
+            StackName=stack_name,
+            PaginationConfig={'MaxItems': events_limit}
```

Comment:
I think it makes sense to only limit the number of events when the `token_filter` isn't passed. That way, if an actual edit was made then all the events resulting from that edit would be returned. CloudFormation has a max number of resource per stack, so there's a natural cap on the number of events per operation. 

What do you think about only having the limit for non-single-op events?

Reply:
I digged a bit more in the code and I don't think that this could help us here.

The [DescribeStackEvents API call](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/API_DescribeStackEvents.html) does not allow any kind of server side filter which would allow for limiting the returned events to a specific stack action. Also the Boto Paginator will need to iterate over all pages first before filtering happens client side when using [JMESPath Filtering](https://boto3.readthedocs.io/en/latest/guide/paginators.html#filtering-results-with-jmespath).

Moreover it seems like the `ClientRequestToken` is generated and passed downstream through all function invocations for every action that is not related to CFN Changesets. So I assume that we would actually end up with a set `token_filter` in the `get_stack_events` function every time.

Am I right with my assumptions?

For us it would also be fine having to configure the limit manually and don't provide a default value. We know that our use case is actually a rare one. 😄

Reply:
Yup, that's sensible to have a default limit since we'll be (typically) filtering out most of the historical events with that ClientRequestToken setting.